logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
I'm a little confused by how this is going to be used. I don't necessarily see why there cannot be 
one synchronized object to keep track of these locks.

In what use case do we have context to the file path, and in what contexts would we just have the Lock?

unless I am mistaken, so far this code is only used in the tests, so I am not sure.